plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -334,6 +335,9 @@ private TypeAndDecoder toTrino(String prefix, IndexMetadata.Field field)
                     return new TypeAndDecoder(VARBINARY, new VarbinaryDecoder.Descriptor(path));
             }
         }
+        else if (type instanceof ScaledFloatType) {
+            return new TypeAndDecoder(DOUBLE, new DoubleDecoder.Descriptor(path));
```

Comment:
can you help understand why we map to `double` and not `real`?

Reply:
That's a good question. Not reason in particular, other than the fact that I haven't been able to find any information in ES documentation about what the expected precision of those values is supposed to be. Let me do some more digging.

Reply:
FWIW, ES accepts values larger than what fit in a `float`, but the results can be very wrong if the scaling internally overflows.

For instance:

```
curl -X PUT "localhost:9200/scaled2?pretty" -H 'Content-Type: application/json' -d '
{
  "mappings": {
    "properties": {
      "foo":    { "type": "scaled_float", "scaling_factor": 1 }
    }
  }
}'

curl -X POST "localhost:9200/scaled2/_doc/?pretty" -H 'Content-Type: application/json' -d' { "foo" : 4e38 } '
```

and then:

```
curl -X GET "localhost:9200/scaled2/_search?pretty" -H 'Content-Type: application/json' -d'
{
      "query" : {
        "match_all" : {
        }
      },
      "_source" : true
}'
```

produces

```json
{
  "hits" : {
    "total" : {
      "value" : 1,
      "relation" : "eq"
    },
    "max_score" : 1.0,
    "hits" : [
      {
        "_index" : "scaled2",
        "_type" : "_doc",
        "_id" : "kw74PHwBKbSBSOQz4da_",
        "_score" : 1.0,
        "_source" : {
          "foo" : 4.0E38
        }
      }
    ]
  }
}
```

And if we do a search:

```
curl -X GET "localhost:9200/scaled2/_search?pretty" -H 'Content-Type: application/json' -d'
{
  "query": {
    "range": {
      "foo": {
        "lte":9223372036854775295
      }
    }
  }
}
'
```

=>

```json
{
  "hits" : {
    "total" : {
      "value" : 0,
      "relation" : "eq"
    },
    "max_score" : null,
    "hits" : [ ]
  }
}
```

but

```
curl -X GET "localhost:9200/scaled2/_search?pretty" -H 'Content-Type: application/json' -d'
{
  "query": {
    "range": {
      "foo": {
        "lte":9223372036854775296
      }
    }
  }
}
'
```

=> 

```json
{
  "hits" : {
    "total" : {
      "value" : 1,
      "relation" : "eq"
    },
    "max_score" : 1.0,
    "hits" : [
      {
        "_index" : "scaled2",
        "_type" : "_doc",
        "_id" : "kw74PHwBKbSBSOQz4da_",
        "_score" : 1.0,
        "_source" : {
          "foo" : 4.0E38
        }
      }
    ]
  }
}

Reply:
thanks for the explanation!